reflex/event.py

```diff
@@ -575,12 +576,32 @@ def download(url: str | Var, filename: Optional[str | Var] = None) -> EventSpec:
         # if filename is not provided, infer it from url
         if filename is None:
             filename = url.rpartition("/")[-1]
+    
+    if url is None:
+        if filename is None:
+            filename = "download.txt"
```

Comment:
is this necessary? up above on line 570, if the filename isn't specified, it's just passed as an empty string.

maybe this whole check could be moved up (outside the `if url is None`) and replace the `if isinstance(url, Var) and filename is None:` bit?

what i'm thinking is remove that check on line 569/570, and move these two lines just above the `if url is None`. That way we'll set `filename` if we have a URL, otherwise, if it remains unspecified, then it can be set to `""` (or `"download.txt"`, i guess, but i'm still not a big fan of making up a name like that)

Reply:
Nice spot, changed this

Reply:
and removed the download.txt to just have "" now

Reply:
turns out having an empty string leads to the download failing on the javascript side as an if statement is not run through to do the download so will stick with download.txt

Reply:
changed so now can be "" again by removing some of the JS code